scripts/md2html/build.sh

```diff
@@ -2,46 +2,81 @@
 
 # Author: @MikeRalphson
 
-# run this script from the root of the repo. It is designed to be run by a GitHub workflow.
+# run this script from the root of the repo
+# It is designed to be run by a GitHub workflow
+
+# Usage: build.sh [version | "latest" | "src"]
+# When run with no arguments, it builds artifacts for all published specification versions.
+# It may also be run with a specific version argument, such as "3.1.1" or "latest"
+# Finally, it may be run with "src" to build "src/oas.md"
+#
 # It contains bashisms
 
-mkdir -p deploy/oas
-mkdir -p deploy/js
+if [ "$1" = "src" ]; then
+  deploydir="deploy-preview"
+else
+  deploydir="deploy/oas"
+fi
+
+mkdir -p $deploydir/js
+mkdir -p $deploydir/temp
+cp -p node_modules/respec/builds/respec-w3c.* $deploydir/js/
 
-cd scripts/md2html
+latest=$(git describe --abbrev=0 --tags)
 
-cp -p ../../node_modules/respec/builds/respec-w3c.* ../../deploy/js/
+if [ -z "$1" ]; then
+  specifications=$(ls -1 versions/[23456789].*.md | grep -v -e "\-editors" | sort -r)
+elif [ "$1" = "latest" ]; then
+  specifications=$(ls -1 versions/$latest.md)
+elif [ "$1" = "src" ]; then
+  specifications="src/oas.md"
+else
+  specifications=$(ls -1 versions/$1.md)
+fi
 
-latest=`git describe --abbrev=0 --tags`
-latestCopied=none
+latestCopied="none"
 lastMinor="-"
-for filename in $(ls -1 ../../versions/[23456789].*.md | sort -r) ; do
-  if [[ ${filename} == *-editors.md ]];then
-    continue
+
+for specification in $specifications; do
+  version=$(basename $specification .md)
+
+  if [ "$1" = "src" ]; then
+    destination="$deploydir/$version.html"
+    maintainers="EDITORS.md"
+  else
+    destination="$deploydir/v$version.html"
+    maintainers="$(dirname $specification)/$version-editors.md"
   fi
 
-  version=$(basename "$filename" .md)
   minorVersion=${version:0:3}
-  tempfile=../../deploy/oas/v$version-tmp.html
-  echo -e "\n=== v$version ==="
+  tempfile="$deploydir/temp/$version.html"
 
-  node md2html.js --maintainers ../../versions/$version-editors.md ${filename} > $tempfile
-  npx respec --use-local --src $tempfile --out ../../deploy/oas/v$version.html
+  echo === Building $version to $destination
+
+  node scripts/md2html/md2html.js --maintainers $maintainers $specification > $tempfile
+  npx respec --use-local --src $tempfile --out $destination
   rm $tempfile
 
+  echo === Built $destination
+
   if [ $version = $latest ]; then
-    if [[ ${version} != *"rc"* ]];then
+    if [[ ${version} != *"rc"* ]]; then
       # version is not a Release Candidate
-      ( cd ../../deploy/oas && ln -sf v$version.html latest.html )
-      latestCopied=v$version
+      ln -sf $(basename $destination) $deploydir/latest.html
+      latestCopied="v$version"
     fi
   fi
 
   if [ ${minorVersion} != ${lastMinor} ] && [ ${minorVersion} != 2.0 ]; then
```

Comment:
```suggestion
  if [ ${minorVersion} != ${lastMinor} ] && [[ ${minorVersion} =~ ^[3-9] ]]; then
```

With argument "src" it now creates a `voas.html` softlink for the "minor version"

Reply:
I'll open a separate PR targeting this behavior.